examples/sonoff_basic/button.c

```diff
@@ -47,9 +48,17 @@ void button_intr_callback(uint8_t gpio) {
     } else {
         // The button is released. Handle the use cases.
         if ((now - button->last_press_time) * portTICK_PERIOD_MS > button->long_press_time) {
```

Comment:
This is kind of simplified version of my original button code (which also supported tracking double presses). Given your additions I suggest to simplify it to just calculating event type based on current time value and then merging all the other code into same one branch.